### DISCUSSION TITLE
Codespaces ping to api.github.com TimedOut

### DISCUSSION THREAD

**Discussion by ldynia** [community/community#48421] [Codespaces]
### Select Topic Area

Bug

### Body

# Context

I'm running github self-hosted runner check script from my codespace.

```bash
./run.sh --check --url https://github.com/$GITHUB_REPOSITORY --pat $SELF_HOSTED_RUNNER_PAT_TOKEN
```

# Problem

The script fails on [Internet Connection Check](https://github.com/actions/runner/blob/main/docs/checks/internet.md) with `ping` command despite correct DNS resolution.

```bash
$ nslookup api.github.com
Server:         127.0.0.53
Address:        127.0.0.53#53

Non-authoritative answer:
Name:   api.github.com
Address: 140.82.121.6

$ nslookup vstoken.actions.githubusercontent.com
Server:         127.0.0.53
Address:        127.0.0.53#53

Non-authoritative answer:
vstoken.actions.githubusercontent.com   canonical name = star-actions-githubusercontent-com.l-0007.l-msedge.net.
star-actions-githubusercontent-com.l-0007.l-msedge.net  canonical name = l-0007.l-msedge.net.
Name:   l-0007.l-msedge.net
Address: 13.107.42.16
Name:   l-0007.l-msedge.net
Address: 2620:1ec:21::16
```

# Error

```bash
Ping api.github.com (0.0.0.0) failed with 'TimedOut'
Ping vstoken.actions.githubusercontent.com (0.0.0.0) failed with 'TimedOut'
```

# Question

The same ping command works from my local network but somehow fails from codespaces? As if something was blocking ping.

# Reproduce

- https://github.com/ldynia/codespace-minikube
- https://stackoverflow.com/questions/75549384/codespaces-ping-to-api-github-com-timed-out

**Comment by sunilsurana** [comment]
Facing same issue, The ping gets stuck. Is there any solution.
